### ISSUE TITLE
[Sonar] Sonarcloud reports 0% coverage even though local reports have coverage

### ISSUE THREAD

**Issue by Decat-SimonA** [kordamp/kordamp-gradle-plugins#335]
Hello,

I have a multi module project that looks like this:
```
app/build.gradle.kts
app/module1/build.gradle.kts
app/module2/build.gradle.kts
```
`module2` has tests that covers classes in `module1`, but `module1` has 0% coverage in Sonarcloud

when I run `gradle build aggregateJacocoReport aggregateJacocoMerge`, the aggregated files is only generated on `app/build/reports/jacoco/aggregate/jacocoTestReport.xml`

but if I add in my `app/build.gradle.kts`:
```
subprojects {
    println(config.coverage.jacoco.aggregateReportXmlFile)
```
it outputs: 
```
/home/foo/dev/project/app/module1/build/reports/jacoco/aggregate/jacocoTestReport.xml
/home/foo/dev/project/app/module2/build/reports/jacoco/aggregate/jacocoTestReport.xml
```

I think the correct output should be:
```
/home/foo/dev/project/app/build/reports/jacoco/aggregate/jacocoTestReport.xml
/home/foo/dev/project/app/build/reports/jacoco/aggregate/jacocoTestReport.xml
```

**Comment by aalmiray** [comment]
This is a side effect of the current DSL. Every project has access to the same config properties, however some are only read from the root project. If you run `gradle :effectiveSettings --section=jacoco` you'll see the settings applied to the jacoco plugin for each project. Note that aggregate settinga are only displayed for the root project.

**Comment by aalmiray** [comment]
I think the empty reports are caused by #336 and #337. did you try using Kordamp `0.36.0`?

**Comment by Decat-SimonA** [comment]
Yes, I am using
```
    id("org.kordamp.gradle.jacoco") version "0.36.0"
    id("org.kordamp.gradle.sonar") version "0.36.0"
```
did you mean `0.38.0` ? (the milestone in #336 and #337)

Tests are run correctly, the aggregate xml and HTML files shows the correct coverage but Sonarcloud still shows "0%" :disappointed:

**Comment by aalmiray** [comment]
No, I did mean `0.36.0` as I know there were some internal changes in `0.37.0` that might have caused the problem, which turned out to be the case after figuring out the fix for #337 (fiddling around with Android support and forgot to remove the comments).

It appears that Jacoco reports are correctly generated but the failure is in the Sonar integration. It may be the case that the Sonar properties set by the `org.kordamp.gradle.sonar` plugin are not the required ones.

**Comment by aalmiray** [comment]
Going back to the original description of this issue. The report generated at the root (whose file points to `rootProject.config.coverage.jacoco.aggregateReportXmlFile`) should contain the aggregate report for all test, integrationTest, and functionalTest tasks. The single `:module2:jacocoTestReport` contains only classes for `module2` even if it depends on `module1`.

If the intention is to exercise coverage on `module1` by running tests in `module2` then explicit configuration in `:module2:jacocoTestReport` must be put in place. This is standard Gradle jacoco configuration, not related to Kordamp.

**Comment by Decat-SimonA** [comment]
> Kordamp does not automatically exercise coverage on dependent projects as the common use case is for module2 to have its own classes and you want to exercise coverage on those classes alone

That makes sense, I though it was done out of the box. A flag to enable it would be a huge lifesaver, I have been blindly copy-pasting solutions from pretty much everywhere and they all seems broken in one way or another

**Comment by ursjoss** [comment]
I have the impression, the issue was introduced with 0.34.0. Sadly I did not notice the coverage drop until I saw the thread of this ticket.

The sonar activity shows the drop with a project with a similar situation:
https://sonarcloud.io/project/activity?graph=coverage&id=ursjoss_KRis

The commit in my project resulting in the coverage decline was https://github.com/ursjoss/KRis/commit/2adbcb48d6232b1af3eb35e5a7bc741854afd7f6 which bumped kordamp from 0.33.0 to 0.34.0.

By just gazing over the commit history of the kordamp-gradle-plugins, I suspect it might have to do with #176, but I don't have time to investigate any further right now.

HTH

**Comment by aalmiray** [comment]
Thank you for the report @ursjoss, this can help me figure out what went wrong since `0.34.0`.

**Comment by aalmiray** [comment]
@Decat-SimonA see #338

**Comment by aalmiray** [comment]
Sample project published with `0.38.0-SNAPSHOT`

https://sonarcloud.io/component_measures?id=kordamp%3Amaster&metric=coverage&view=list

**Comment by ursjoss** [comment]
> @ursjoss this project is based on your reproducers (01_java_gradle_groovy). As you can see in the report there's no sign of Integration nor Functional tests, just unit tests, although all tests were executed locally.

@aalmiray Thanks for the hint, I missed the misconfiguration on that part.
